src/watcher.rs

```
@@ -19,5 +43,15 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Sync + Send>> {
         }
         // Wait for X milliseconds before fetching events again
         sleep(Duration::from_millis(config.watcher_sleep)).await;
+        retry_failed_events += 1;
+
+        if RETRY_THRESHOLD == retry_failed_events {
+            let sender = event_processor.sender.lock().await;
+            let _ = sender
+                .send(SenderMessage::RetryEvent(
```

why the `let _`?  
This will fail silently if the `.send` fails and essentially breaks the retry manager.  
It's better to have this throw the exception if necessary.

---

yep, might be better approach but now it does not exist (deleted already)